Can select children of instanced scene while it doesn't have editable children

**Godot version:**
3.2.4.rc.custom_build / 3.2.4.rc1 (both exhibit the same behaviour)

**System**
Linux (PopOS - most recent, up to date including drivers), GLES3, Nvidia 1060 - laptop

**Issue description:**
This is a really weird bug. It only affects projects created with previous versions of Godot.
As of rc1 i believe, the 3d editor unreliably alters between selecting the instanced scene or its children.
No errors whatsoever. In the screenshot below you can see:
- instanced scene which doesnt have editable children
- first click on the scene selected one of its children
- understandably the node is not highlighted in the scene tree as it's not there at all.

![image](https://user-images.githubusercontent.com/38346332/107890154-a8b07680-6f17-11eb-8c04-a9979678bb6e.png)


**Steps to reproduce:**
Open a project created in 3.2.3 or lower

**Minimal reproduction project:**
Sadly I am not in possession of a minimal reproduction project, I've only got bigger ones, which I don't mind sharing if it'd help and you don't mind downloading a bigger project.

@hilfazer super sorry, I pulled your branch and went to build godot, sadly I got ALSA related errors that I had never met before. Not sure how to fix them either, meaning it looks like I wont be able to test your fix! :(
```bash
[100%] /usr/include/alsa/rawmidi.h:82: error: variable 'snd_rawmidi_open' redeclared as function
   82 | int snd_rawmidi_open(snd_rawmidi_t **in_rmidi, snd_rawmidi_t **out_rmidi,
      | 
drivers/alsa/asound-so_wrap.c:3042: note: previously declared here
 3042 | int (*snd_rawmidi_open)( snd_rawmidi_t**, snd_rawmidi_t**,const char*, int);
      | 
/usr/include/alsa/rawmidi.h:86: error: variable 'snd_rawmidi_close' redeclared as function
   86 | int snd_rawmidi_close(snd_rawmidi_t *rmidi);
      | 
drivers/alsa/asound-so_wrap.c:3044: note: previously declared here
 3044 | int (*snd_rawmidi_close)( snd_rawmidi_t*);
...
```

--- a/editor/plugins/spatial_editor_plugin.cpp
+++ b/editor/plugins/spatial_editor_plugin.cpp
@@ -75,6 +75,20 @@
 #define MIN_FOV 0.01
 #define MAX_FOV 179
 
+static Node *get_deepest_visible_node(Node *start_node, Node const *edited_scene) {
+	Node const *iterated_item = start_node;
+	Node *node = start_node;
+
+	while (iterated_item->get_owner() && iterated_item->get_owner() != edited_scene) {
+		if (!edited_scene->is_editable_instance(iterated_item->get_owner()))
+			node = iterated_item->get_owner();
+
+		iterated_item = iterated_item->get_owner();
+	}
+
+	return node;
+}
+
 void ViewportRotationControl::_notification(int p_what) {
 
 	if (p_what == NOTIFICATION_ENTER_TREE) {
@@ -532,11 +546,7 @@ ObjectID SpatialEditorViewport::_select_ray(const Point2 &p_pos, bool p_append, 
 			continue;
 
 		if (dist < closest_dist) {
-
-			item = Object::cast_to<Node>(spat);
-			while (item->get_owner() && item->get_owner() != edited_scene && !edited_scene->is_editable_instance(item->get_owner())) {
-				item = item->get_owner();
-			}
+			item = get_deepest_visible_node(Object::cast_to<Node>(spat), edited_scene);
 
 			closest = item->get_instance_id();
 			closest_dist = dist;
@@ -691,10 +701,7 @@ void SpatialEditorViewport::_select_region() {
 		if (!sp || _is_node_locked(sp))
 			continue;
 
-		Node *item = Object::cast_to<Node>(sp);
-		while (item->get_owner() && item->get_owner() != edited_scene && !edited_scene->is_editable_instance(item->get_owner())) {
-			item = item->get_owner();
-		}
+		Node *item = get_deepest_visible_node(Object::cast_to<Node>(sp), edited_scene);
 
 		// Replace the node by the group if grouped
 		if (item->is_class("Spatial")) {
@@ -1026,7 +1033,7 @@ void SpatialEditorViewport::_list_select(Ref<InputEventMouseButton> b) {
 
 	for (int i = 0; i < selection_results.size(); i++) {
 		Spatial *item = selection_results[i].item;
-		if (item != scene && item->get_owner() != scene && !scene->is_editable_instance(item->get_owner())) {
+		if (item != scene && item->get_owner() != scene && item != get_deepest_visible_node(item, scene)) {
 			//invalid result
 			selection_results.remove(i);
 			i--;

